Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docrepository/adapter/AdapterComponentDocRepositoryOrchImpl.java

```diff
@@ -274,6 +230,80 @@ protected void loadDocumentResponses(ihe.iti.xds_b._2007.RetrieveDocumentSetResp
         }
     }
 
+    /**
+     * Creates a document response for each document and adds it to the document response list
+     *
+     * Sets the Document Unique Id, HCID, Repository Unique Id, and Mime Type. If the document is an On-Demand Document,
+     * the NewDocument UniqueId and Repository Unique Id will also be set.
+     *
+     * Finally, the document metadata will be added to the list if, and only if, the document metadata contains a valid
+     * document
+     *
+     * @param docs - List of document metadata to create responses from
+     * @param homeCommunityId
+     * @param docResponseList - The container that will hold the generated responses
+     */
+    private void createDocResponses(List<DocumentMetadata> docs, String homeCommunityId,
+        List<DocumentResponse> docResponseList) {
+        for (DocumentMetadata doc : docs) {
+            DocumentResponse oDocResponse = new DocumentResponse();
+
+
+            oDocResponse.setHomeCommunityId(homeCommunityId);
+            oDocResponse.setRepositoryUniqueId(REPOSITORY_UNIQUE_ID);
+
+            // Document Unique ID
+            if (NullChecker.isNotNullish(doc.getDocumentUniqueId())) {
+                oDocResponse.setDocumentUniqueId(doc.getDocumentUniqueId());
+                LOG.debug("Document unique id found ");
+            }
+
+            // Mime Type
+            if (NullChecker.isNotNullish(doc.getMimeType())) {
+                oDocResponse.setMimeType(doc.getMimeType());
+                LOG.debug("Mime type Identified ");
+            }
+
+            // On-Demand document
+            if (doc.isOnDemand()) {
+                oDocResponse.setNewDocumentUniqueId(doc.getNewDocumentUniqueId());
+                oDocResponse.setNewRepositoryUniqueId(doc.getNewRepositoryUniqueId());
+            }
+
+            if (setDocumentResponse(doc, oDocResponse)) {
+                docResponseList.add(oDocResponse);
+            }
+        }
+    }
+
+    /**
```

Comment:
Optional: add descript to what this method do and what it returns

Reply:
Done.